src/rail/estimation/algos/lephare.py

```diff
@@ -206,52 +196,104 @@ def _process_chunk(self, start, end, data, first):
-            self.photz.prep_data(oneObj)
-            srclist.append(oneObj)
-
-        # Run autoadapt to improve zero points
-        a0, a1 = self.photz.run_autoadapt(srclist)
-        offsets = ",".join(np.array(a0).astype(str))
-        offsets = "# Offsets from auto-adapt: " + offsets + "\n"
-        print(offsets)
-
-        photozlist = []
-        for i in range(ng):
-            oneObj = lp.onesource(i, self.photz.gridz)
-            oneObj.readsource(str(i), flux[i], flux_err[i], 63, zspec[i], " ")
-            self.photz.prep_data(oneObj)
-            photozlist.append(oneObj)
-
-        self.photz.run_photoz(photozlist, a0, a1)
 
         for i in range(ng):
-            pdf, zgrid = self._estimate_pdf(photozlist[i])
-            pdfs.append(pdf)
             # Take median in case multiple probability densities are equal
             zmode[i] = np.median(
                 zgrid[np.where(pdfs[i] == np.max(pdfs[i]))[0].astype(int)]
             )
             zmean[i] = (zgrid * pdfs[i]).sum() / pdfs[i].sum()
 
         qp_dstn = qp.Ensemble(qp.interp, data=dict(xvals=zgrid, yvals=np.array(pdfs)))
-        qp_dstn.set_ancil(dict(zmode=zmode, zmean=zmean))
+        ancil = dict(zmode=zmode, zmean=zmean)
+        # Add the requested outputs.
+        for c in self.config["output_keys"]:
+            ancil[c] = np.array(output[c])
+        qp_dstn.set_ancil(ancil)
         self._do_chunk_output(qp_dstn, start, end, first)
+
+
+def _rail_to_lephare_input(data, mag_cols, mag_err_cols):
+    """Take the rail data input and convert it to that expected by lephare
+
+    Parameters
+    ==========
+    data : pandas
+        The RAIL input data chunk
+
+    Returns
+    =======
+    input : astropy.table.Table
+        The lephare input
+
+
+    """
+    ng = data["redshift"].shape[0]
+    # Make input catalogue in standard lephare format
+    input = Table()
+    try:
+        input["id"] = data["id"]
+    except KeyError:
+        input["id"] = np.arange(ng)
+    # Add all available magnitudes
+    for n in np.arange(len(mag_cols)):
+        input[mag_cols[n]] = data[mag_cols[n]].T
+        input[mag_err_cols[n]] = data[mag_err_cols[n]].T
+    # Set context to use all bands
+    input["context"] = np.sum([2**n for n in np.arange(ng)])
+    input["zspec"] = data["redshift"]
+    input["string_data"] = " "
+    return input
+
+
+def _update_lephare_env(lepharedir, lepharework):
+    """Update the environment variables and reset the lephare package.
+
+    We may be using the same Python session to run inform with different
+    settings. These produce intermediate files which are distinct and we must
+    use different runs.
+
+    This simple function updates the environment variables and reloads lephare
+    to ensure they are properly used.
+    """
+    if lepharedir is not None:
+        os.environ["LEPHAREDIR"] = lepharedir
+    if lepharework is not None:
+        os.environ["LEPHAREWORK"] = lepharework
+    importlib.reload(lp)
```

Comment:
If this is working, then that's fine, I don't think it's worth poking a bear. But I really do wish we had a way around this that didn't require it.